plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt

```diff
@@ -36,6 +39,27 @@ class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
             }
         )
 
+        project.messageBus.connect().subscribe(
+            ToolWindowManagerListener.TOPIC,
+            object : ToolWindowManagerListener {
+                override fun stateChanged(
+                    toolWindowManager: ToolWindowManager,
+                    toolWindow: ToolWindow,
+                    changeType: ToolWindowManagerListener.ToolWindowManagerEventType
+                ) {
+                    if (changeType == ToolWindowManagerListener.ToolWindowManagerEventType.MovedOrResized) {
+                        val width = toolWindow.component.width
+                        if (width < MINIMUM_TOOLWINDOW_WIDTH) {
+                            LOG.debug {
+                                "Amazon Q Tool window stretched to a width less than the minimum allowed width, resizing to the minimum allowed width"
+                            }
+                            (toolWindow as ToolWindowEx).stretchWidth(MINIMUM_TOOLWINDOW_WIDTH - width)
+                        }
+                    }
+                }
+            }
+        )
+
```

Comment:
can we just add a ComponentListener directly to the underlying webview component and react to `componentResized`

Reply:
done